### PR TITLE
global: refactoring of models

### PR DESCRIPTION
**Pull request by lnielsen**
* Refactors models to support sharing the same files between multiple buckets.

Signed-off-by: Lars Holm Nielsen <lars.holm.nielsen@cern.ch>

### CODE REVIEW

**Comment by PXke** [comment]
Looks good

File: invenio_files_rest/config.py [hunk 1]
```diff
@@ -22,12 +22,21 @@
 # waive the privileges and immunities granted to it by virtue of its status
 # as an Intergovernmental Organization or submit itself to any jurisdiction.
 
-"""Invenio File Rest module configuration file."""
+"""Invenio Files Rest module configuration file."""
 
 FILES_REST_STORAGE_CLASS_LIST = {
```
Comment (jirikuncar) [inline invenio_files_rest/config.py]: IMHO it is not configuration but it should be defined as enum in models. Can admin define more storage classes? If yes, then we should have something like: `'S': 'invenio_files_rest.storage:Standard'`.

File: invenio_files_rest/models.py [hunk 2]
```diff
@@ -157,92 +249,120 @@ class Object(db.Model, Timestamp):
...
-                cls.bucket_id == bucket_id,
-                ~cls.deleted,
-                cls.updated == db.session.query(
-                    db.func.max(table_object.updated)).filter(
-                        ~table_object.deleted,
-                        table_object.filename == cls.filename).limit(1)
-            )
+        args = [
+            cls.bucket_id == bucket_id,
+            cls.file_id != None,  # noqa
```
Comment (jirikuncar) [inline invenio_files_rest/models.py]: `cls.file_id.isnot(None),`

File: invenio_files_rest/config.py (hunk 1 above)
Comment (lnielsen) [inline invenio_files_rest/config.py]: The idea behind storage class is to allow a system to record what type of storage a given file resides on - e.g. offline/online, reduced resilience, ... . In Files-REST there's no specific functionality associated with it, but an Invenio instance may use it to adapt behaviour depending no the storage class.

File: invenio_files_rest/models.py (hunk 2 above)
Comment (lnielsen) [inline invenio_files_rest/models.py]: Thanks

File: invenio_files_rest/views.py [hunk 3]
```diff
@@ -490,15 +412,91 @@ def get(self, bucket_id, filename, **kwargs):
...
+                {
+                    "checksum": "xxxx",
+                    "size": 0,
+                    "url": "http://localhost:5000/files/322ea4c6-665.../f.pdf",
+                    "uuid": "322ea4c6-6650-4143-a328-274eee55f45d",
+                    "updated": "2015-12-10T14:16:57.202795"
+                }
+
+            :resheader Content-Type: application/json
+            :statuscode 200: no error
```
Comment (blixhavn) [inline invenio_files_rest/views.py]: Shouldn't this be 201/202?

File: tests/conftest.py [hunk 4]
```diff
@@ -48,53 +48,45 @@ def app(request):
...
-    Babel(app_)
-    InvenioFilesREST(app_)
-
-    app_.config['SQLALCHEMY_DATABASE_URI'] = os.environ.get(
-        'SQLALCHEMY_DATABASE_URI', 'sqlite:///test.db'
+        TESTING=True,
+        SQLALCHEMY_TRACK_MODIFICATIONS=True,
+        SQLALCHEMY_DATABASE_URI=os.environ.get(
+            'SQLALCHEMY_DATABASE_URI',
+            'postgresql+psycopg2://localhost:5432/files'),
```
Comment (ddaze) [inline tests/conftest.py]: Not sqlite as default?

**Pull request merged**